### PR TITLE
Added $Minimum Engine Volume to ships.tbl, with which to customize ho…

### PR DESCRIPTION
**Pull request by ln-zookeeper**
…w the ship's speed affects engine sound volume.

### CODE REVIEW

File: code/ship/ship.cpp [hunk 1]
```diff
@@ -2665,6 +2666,9 @@ int parse_ship_values(ship_info* sip, bool first_time, bool replace)
 	//Parse the engine sound
 	parse_sound("$EngineSnd:", &sip->engine_snd, sip->name);
 
+	if(optional_string("$Minimum Engine Volume:"))
+		stuff_float(&sip->min_engine_vol);
```
Comment (asarium) [inline code/ship/ship.cpp]: Validation of the user supplied value would be nice given that only values between `0.0f` and `1.0f` make sense.

File: code/object/objectsnd.cpp [hunk 2]
```diff
@@ -521,17 +521,24 @@ void obj_snd_do_frame()
...
 					percent_max = 0.0f;
 				}
 				else
 					percent_max = objp->phys_info.fspeed / objp->phys_info.max_vel.xyz.z;
 
-				if ( percent_max >= 0.5f )
-					speed_vol_multiplier = 1.0f;
-				else {
-					speed_vol_multiplier = 0.5f + (percent_max);	// linear interp: 0.5->1.0 when 0.0->0.5
+				if ( sip->min_engine_vol == -1.0f) {
```
Comment (asarium) [inline code/object/objectsnd.cpp]: The comments in this section don't match what the code does (if I read it correctly).

I think for retail the following graph applies:
msp11151c01ig5i2eab464a00002h1i18g9786f4c74
The volume interpolates between 0.5 and 1 when the ship is between 0 and 50% of its speed. Everything above is capped at 1.


The new option results in the following graph (for `min_engine_vol` = 0.5):
msp31521d648hf025a1ba60000013a9b94bb92762c0
I think either linearly interpolating between `min_engine_vol` and `1.0f` based on the speed percentage or following the retail code would be more appropriate here.

File: code/ship/ship.cpp (hunk 1 above)
Comment (ln-zookeeper) [inline code/ship/ship.cpp]: Actually I specifically chose not to validate it, in order to not unnecessarily prevent one from doing something weird and unusual with it. You could for example use values >1.0 to make the sound volume drop as speed increases, which could have unexpected uses for, say, an otherworldly alien probe.

Comment (asarium) [inline code/ship/ship.cpp]: Ahh, the formula would need to be documented, then users can figure out how different values would change the behavior.

**Comment by chief1983** [comment]
With the volume ramping adjusted, what else is this waiting on?  Code or wiki documentation on the formula?

**Comment by asarium** [comment]
Apparently there are some merge conflicts, once those are fixed this can be merged.

**Pull request merged**